ttnn/CMakeLists.txt

```diff
@@ -59,28 +61,36 @@ set(ttnn_lib_include_dirs
     ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/experimental/tensor
     ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/experimental/queue
     ${CMAKE_CURRENT_SOURCE_DIR}/cpp
-    ${PROJECT_SOURCE_DIR}/tt_metal/third_party/fmt
-)
+    ${PROJECT_SOURCE_DIR}/tt_metal/third_party/fmt)
+set(TTNN_PUBLIC_LINK_LIBRARIES compiler_flags metal_header_directories metal_common_libs tt_metal linker_flags) # linker_flags = -rdynamic if tracy enabled
+set(TTNN_PUBLIC_LINK_DIRS "")
 
-add_library(ttnn_lib OBJECT ${TTNN_SRCS})
-target_compile_options(ttnn_lib PUBLIC ${ttnn_compile_options})
-target_link_libraries(ttnn_lib PUBLIC ${ttnn_link_libraries})
-target_include_directories(ttnn_lib PUBLIC ${ttnn_lib_include_dirs})
-target_precompile_headers(ttnn_lib PRIVATE
+set(TTNN_PRECOMPILED_HEADERS
     ${PROJECT_SOURCE_DIR}/tt_metal/third_party/magic_enum/magic_enum.hpp
+    ${PROJECT_SOURCE_DIR}/tt_metal/third_party/tracy/public/tracy/Tracy.hpp
```

Comment:
would adding this many new pch files introduce more risk for incremental compile issues?

Reply:
I suggest to remove all non-third-party headers from this list. They mostly don't matter, but they force the PCHs to recompile each time we change them.

Reply:
This is the exact list used for tt_dnn before my changes.
I agree, will adjust.